contrib/kube-prometheus/examples/prometheus-pvc.jsonnet

```diff
@@ -0,0 +1,59 @@
+
+// Reference info: documentation for https://github.com/ksonnet/ksonnet-lib can be found at http://g.bryan.dev.hepti.center
+//
+local k = import 'ksonnet/ksonnet.beta.3/k.libsonnet'; // https://github.com/ksonnet/ksonnet-lib/blob/master/ksonnet.beta.3/k.libsonnet - imports k8s.libsonnet
+// * https://github.com/ksonnet/ksonnet-lib/blob/master/ksonnet.beta.3/k8s.libsonnet defines things such as "persistentVolumeClaim:: {"
+//
+local pvc = k.core.v1.persistentVolumeClaim; // https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#persistentvolumeclaim-v1-core (defines variable named 'spec' of type 'PersistentVolumeClaimSpec')
+
+local kp =
+  (import 'kube-prometheus/kube-prometheus.libsonnet') +
+  (import 'kube-prometheus/kube-prometheus-bootkube.libsonnet') +
+  {
+    _config+:: {
+      namespace: 'monitoring',
+    },
+
+    prometheus+:: {
+      prometheus+: {
+        spec+: { // https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#prometheusspec
+          // If a value isn't specified for 'retention', then by default the '--storage.tsdb.retention=24h' arg will be passed to prometheus by prometheus-operator.
+          // The possible values for a prometheus <duration> are:
+          //  * https://github.com/prometheus/common/blob/c7de230/model/time.go#L178 specifies "^([0-9]+)(y|w|d|h|m|s|ms)$" (years weeks days hours minutes seconds milliseconds)
+          retention: "30d",
+
+          // Reference info: https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/storage.md
+          // By default (if the following 'storage.volumeClaimTemplate' isn't created), prometheus will be created with an EmptyDir for the 'prometheus-k8s-db' volume (for the prom tsdb).
+          // This 'storage.volumeClaimTemplate' causes the following to be automatically created (via dynamic provisioning) for each prometheus pod:
+          //  * PersistentVolumeClaim (and a corresponding PersistentVolume)
+          //  * the actual volume (per the StorageClassName specified below)
+          storage: { // https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#storagespec
```

Comment:
using this, when i deployed k-p to a K8s cluster in AWS, the Name of the AWS EBS Volume that was automatically created starts with "kubernetes-dynamic-pvc-" (and then has a bunch of random/generated letters & digits). Can anyone (perhaps @brancz ) tell me how to get a given specified string of text to be included somewhere in the EBS Volume? (The company I work for would like to be able to easily identify which of our teams are creating/using which cloud resources.)

More info:

I tried adding the following line...:
`pvc.mixin.metadata.withGenerateName('my-team-name'),`

...but the EBS that was created still had the same generic name format - and the only difference i noticed was the following (note that "generateName: my-team-name" is under the "metadata" - but "my-team-name" doesn't show up anywhere in the EBS Volume):
```
core@ip-192-168-xx-xx ~ $ kubectl -n monitoring get pvc prometheus-k8s-db-prometheus-k8s-0 -o yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    pv.kubernetes.io/bind-completed: "yes"
    pv.kubernetes.io/bound-by-controller: "yes"
    volume.beta.kubernetes.io/storage-provisioner: kubernetes.io/aws-ebs
  creationTimestamp: 2018-08-30T14:34:13Z
  finalizers:
  - kubernetes.io/pvc-protection
  generateName: my-team-name
  labels:
    app: prometheus
    prometheus: k8s
  name: prometheus-k8s-db-prometheus-k8s-0
  namespace: monitoring
  resourceVersion: "3637001"
  selfLink: /api/v1/namespaces/monitoring/persistentvolumeclaims/prometheus-k8s-db-prometheus-k8s-0
  uid: c3f40e8d-ac61-11e8-a38a-1288318eea14
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 100Gi
  storageClassName: ssd
  volumeName: pvc-c3f40e8d-ac61-11e8-a38a-1288318eea14
status:
  accessModes:
  - ReadWriteOnce
  capacity:
    storage: 100Gi
  phase: Bound
```

(I Googled, & [this](https://github.com/ksonnet/parts/blob/master/incubator/redis/redis.libsonnet) is the only e.g. I could find of [ksonnet persistentVolumeClaim](http://g.bryan.dev.hepti.center/core/v1/persistentVolumeClaim/#core.v1.persistentVolumeClaim) being used.)

Reply:
Through the PersistentVolume you can find which PersistentVolumeClaim has claimed it, that way you can make the correlation between namespace/object name and the team it belongs to.